thingooConnector/thingooconnector.py

```diff
@@ -0,0 +1,40 @@
+from thingooConnector.connector import Connector
+
+from thingooConnector.httpconnector import HTTPConnector
+from thingooConnector.mqttconnector import MQTTConnector
+
+
+class ThingooConnector(Connector):
+    def __init__(self, host, device_info, entities, client_credentials, mqtt_credentials=None, http_only=False):
+        """
+        :param host: Thingoo host
+        :param device_info: A device info object
+        :type device_info: :class:`DeviceInfo`
+        :param entities: A list of entities
+        :param client_credentials: A ClientCredentials object used to get access_token
+        :type device_info: :class:`ClientCredentials`
+        :param mqtt_credentials: A UserData object used to connect to MQTT
+        :type mqtt_credentials: :class:`MQTTCredentials`
+        :param http_only: A boolean value that indicates if the device should only use http without connecting to mqtt
+        :type http_only: bool
+        """
+        super().__init__(host, device_info, entities)
+        self._http_connector = HTTPConnector(host, device_info, entities, client_credentials)
+        self._http_only = http_only
```

Comment:
saving this into a class field feels redundant - things can be simplified by only initializing the `MQTTConnector` if `http_only` is false and replacing the `if not self._http_only` checks below with a `if self._mqtt_connector`, for example.

Reply:
I guess there is more logic built on top of http_only. A device will use http for readings if this flag is set to true: 
see ThingooConnector.publish_entity_reading

Reply:
That sounds good to me, apart from that currently, we're deciding whether to create `self._mqtt_connector` based on the `_http_only` flag. Should we instead use `mqtt_credentials` for checking if the user also wants to use mqtt beside http?

Reply:
I'm not suggesting to remove the `http_only` flag entirely - just to avoid saving it for later, since if it only guards creating one object in the constructor, we can easily replace it everywhere else with a check for the said object, which the getter would perform.

Reply:
Resolved in 57ae1c8